.github/workflows/rust.yml

```diff
@@ -340,6 +340,29 @@ jobs:
           --mode validate \
           -c envoy.yaml
 
+  doc-test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install dependencies
+      if: ${{ env.ACT }}
+      run: |
+        curl -OL https://static.rust-lang.org/rustup/rustup-init.sh
+        chmod +x ./rustup-init.sh
+        ./rustup-init.sh -y
+        rm rustup-init.sh
+        echo "$HOME/.cargo/bin" >> $GITHUB_PATH
+
+    - name: Update Rust
+      run: |
+        rustup toolchain install stable --component clippy --component rustfmt
+        rustup target add wasm32-wasi
+
+    - name: Run Cargo Doc Test
+      run: cargo test --doc
```

Comment:
This should be added as an extra step into `msrv`, `stable` and `nightly` jobs.

Reply:
Updated.